tools/accuracy_checker/custom_evaluators/colorization_v2_evaluator.py

```diff
@@ -0,0 +1,215 @@
+"""
+Copyright (c) 2019 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import numpy as np
+from accuracy_checker.evaluators.base_evaluator import BaseEvaluator
+from accuracy_checker.dataset import Dataset
+from accuracy_checker.adapters import create_adapter
+from accuracy_checker.data_readers import BaseReader
+from accuracy_checker.config import ConfigError
+from accuracy_checker.preprocessor import PreprocessingExecutor
+from accuracy_checker.metrics import MetricsExecutor
+from accuracy_checker.launcher import create_launcher
+from accuracy_checker.utils import extract_image_representations
+import cv2
+
+class ColorizationEvaluator(BaseEvaluator):
+    def __init__(self, dataset, reader, preprocessing, metric_executor, launcher, test_model, check_model):
+        self.dataset = dataset
+        self.preprocessing_executor = preprocessing
+        self.metric_executor = metric_executor
+        self.launcher = launcher
+        self.test_model = test_model
+        self.check_model = check_model
+        self.reader = reader
+        self._metrics_results = []
+
+    @classmethod
+    def from_configs(cls, config):
+        dataset_config = config['datasets'][0]
+        dataset = Dataset(dataset_config)
+        data_reader_config = dataset_config.get('reader', 'opencv_imread')
+        data_source = dataset_config['data_source']
+        if isinstance(data_reader_config, str):
+            reader = BaseReader.provide(data_reader_config, data_source)
+        elif isinstance(data_reader_config, dict):
+            reader = BaseReader.provide(data_reader_config['type'], data_source, data_reader_config)
+        else:
+            raise ConfigError('reader should be dict or string')
+        preprocessing = PreprocessingExecutor(dataset_config.get('preprocessing', []), dataset.name)
+        metrics_executor = MetricsExecutor(dataset_config['metrics'], dataset)
+        launcher = create_launcher(config['launchers'][0], delayed_model_loading=True)
+        test_model = ColorizationTestModel(config.get('network_info', {}), launcher)
+        check_model = ColorizationCheckModel(config.get('network_info', {}),
+                                           dataset_config.get('preprocessing', {}), launcher)
```

Comment:
preprocessing in config is list, not dictionary.

Reply:
@eaidova done)